lib/functions/main/build-tasks.sh

```diff
@@ -0,0 +1,281 @@
+
+		[[ "${REPOSITORY_INSTALL}" != *armbian-config* ]] && compile_armbian-config
+
+	fi
+}
+
+build_armbian-zsh() {
+	# Compile armbian-zsh if packed .deb does not exist or use the one from repository
+	if [[ ! -f ${DEB_STORAGE}/armbian-zsh_${REVISION}_all.deb ]]; then
+
+		[[ "${REPOSITORY_INSTALL}" != *armbian-zsh* ]] && compile_armbian-zsh
+
+	fi
+}
+
+build_plymouth-theme-armbian() {
+	# Compile plymouth-theme-armbian if packed .deb does not exist or use the one from repository
+	if [[ ! -f ${DEB_STORAGE}/plymouth-theme-armbian_${REVISION}_all.deb ]]; then
+
+		[[ "${REPOSITORY_INSTALL}" != *plymouth-theme-armbian* ]] && compile_plymouth-theme-armbian
+
+	fi
+}
+
+build_armbian-firmware() {
+	# Compile armbian-firmware if packed .deb does not exist or use the one from repository
+	if ! ls "${DEB_STORAGE}/armbian-firmware_${REVISION}_all.deb" 1> /dev/null 2>&1 || ! ls "${DEB_STORAGE}/armbian-firmware-full_${REVISION}_all.deb" 1> /dev/null 2>&1; then
+
+		if [[ "${REPOSITORY_INSTALL}" != *armbian-firmware* ]]; then
+			[[ "${INSTALL_ARMBIAN_FIRMWARE:-yes}" == "yes" ]] && { # Build firmware by default.
+				FULL=""
+				REPLACE="-full"
+				compile_firmware
+				FULL="-full"
+				REPLACE=""
+				compile_firmware
+			}
+
+		fi
+
+	fi
+}
+
+build_armbian-bsp() {
+	# create board support package
+	[[ -n "${RELEASE}" && ! -f "${DEB_STORAGE}/${BSP_CLI_PACKAGE_FULLNAME}.deb" && "${REPOSITORY_INSTALL}" != *armbian-bsp-cli* ]] && create_board_package
+
+	# create desktop package
+	[[ -n "${RELEASE}" && "${DESKTOP_ENVIRONMENT}" && ! -f "${DEB_STORAGE}/$RELEASE/${CHOSEN_DESKTOP}_${REVISION}_all.deb" && "${REPOSITORY_INSTALL}" != *armbian-desktop* ]] && create_desktop_package
+	[[ -n "${RELEASE}" && "${DESKTOP_ENVIRONMENT}" && ! -f "${DEB_STORAGE}/${RELEASE}/${BSP_DESKTOP_PACKAGE_FULLNAME}.deb" && "${REPOSITORY_INSTALL}" != *armbian-bsp-desktop* ]] && create_bsp_desktop_package
+}
+
+build_chroot() {
+	# build additional packages
+	[[ $EXTERNAL_NEW == compile ]] && chroot_build_packages
+}
+
+build_bootstrap() {
+	# These two keys are necessary for backward compatibility with logic
+	# https://github.com/armbian/scripts/tree/master/.github/workflows scripts.
+	# They need to be removed when the need disappears there.
+	if [[ $KERNEL_ONLY != yes ]]; then
+		[[ $BSP_BUILD != yes ]] && debootstrap_ng
+	fi
+}
+
+#################################################################################################################################
+#
+# build_main()
+#
+# Builds all artifacts or the filtered ones only.
+# Ensures that any build pre-requisite is met.
+#
+# $1: optional comma separated list of artifacts to build only.
+#     If this list is empty or not set, then all build tasks will be performed.
+#     The following build task names are supported for filtering build tasks:
+#       u-boot, kernel, armbian-config, armbian-zsh, plymouth-theme-armbian, armbian-firmware, armbian-bsp, chroot, bootstrap
+#
+build_main() {
```

Comment:
I don't think `build_main` should require arguments. Because the sub-function will use `BUILD_ONLY` directly. So passing `BUILD_ONLY` as argument may cause confused.

Reply:
Clear - see answer below - will take care of it.